torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3189,6 +3211,23 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
            dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
            sample_inputs_func=sample_inputs_masked_select),
+    OpInfo('matmul',
+           dtypes=floating_types(),
+           dtypesIfCPU=all_types_and_complex(),
+           dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
+                                           *[torch.bfloat16] if CUDA11OrLater else []),
+           dtypesIfROCM=floating_types_and(torch.half),
+           assert_autodiffed=True, 
+           sample_inputs_func=sample_inputs_matmul,
+           skips=(
+               # matmul does not correctly warn when resizing out= inputs
+               SkipInfo('TestCommon', 'test_out'),
+               # matmul on complex128 doesn't work
+               SkipInfo('TestGradients', 'test_fn_grad',
+                        device_type='cpu', dtypes=(torch.complex128,)),
+               # some how it doesn't throw on cpu, and crash when enabled 
+               SkipInfo('TestOpInfo', 'test_unsupported_dtypes', 
+                        device_type='cpu', dtypes=(torch.float16,)),)),
```

Comment:
@mruberry which one of these skip is worth filing an issue? 

I think `test_unsupported_dtypes` should definitely be one since neither support nor unsupport passes for float16. 
not sure about the other 2 though.